src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -6,295 +6,192 @@
     }
   }
 
-  private static void drawWithTranslate(final Graphics2D graphics, final Shape shape, final double translateX,
-      final double translateY) {
-    if (shape instanceof Ellipse2D.Double) {
-      Ellipse2D.Double elipse = (Ellipse2D.Double) shape;
-      elipse = new Ellipse2D.Double(elipse.x + translateX, elipse.y + translateY, elipse.width, elipse.height);
-      graphics.draw(elipse);
+  private static void drawJoints(Graphics2D graphics, Point[] points, int xOffset, int yOffset, int jointsize,
+      double scale) {
+    for (Point p : points) {
+      graphics.fillOval((int) (((p.x - xOffset) - jointsize / 2) * scale),
+          (int) (((p.y - yOffset) - jointsize / 2) * scale), jointsize, jointsize);
     }
-    if (shape instanceof Polygon) {
-      ((Polygon) shape).translate((int) translateX, (int) translateY);
-      graphics.fill(shape);
-      ((Polygon) shape).translate((int) -translateX, (int) -translateY);
+
+  }
+
+  protected static double[] getIndex(Point[] points) {
+    final double[] index = new double[points.length];
+    if(index.length > 0){
+      index[0] = 0;
     }
-    if (shape instanceof Line2D) {
-      final Line2D line = (Line2D) shape;
-      final Point2D p1 = new Point2D.Double(line.getP1().getX() + translateX, line.getP1().getY() + translateY);
-      final Point2D p2 = new Point2D.Double(line.getP2().getX() + translateX, line.getP2().getY() + translateY);
-      graphics.draw(new Line2D.Double(p1, p2));
+    for (int i = 1; i < points.length; i++) {
+      index[i] = index[i - 1] + Math.sqrt(points[i - 1].distance(points[i]));
     }
-    if (shape instanceof QuadCurve2D) {
-      QuadCurve2D.Double curve = (QuadCurve2D.Double) shape;
-      curve = new QuadCurve2D.Double(curve.x1 + translateX, curve.y1 + translateY, curve.ctrlx + translateX,
-          curve.ctrly + translateY, curve.x2 + translateX, curve.y2 + translateY);
-      graphics.draw(curve);
+    return index;
+  }
+
+  private static void drawLineWithTranslate(Graphics2D graphics, Line2D line, double translateX, double translateY,
+      double scale) {
+    final Point2D point1 =
+        new Point2D.Double((line.getP1().getX() - translateX) * scale, (line.getP1().getY() - translateY) * scale);
+    final Point2D point2 =
+        new Point2D.Double((line.getP2().getX() - translateX) * scale, (line.getP2().getY() - translateY) * scale);
+    graphics.draw(new Line2D.Double(point1, point2));
+  }
+
+  protected static Point[] getRoutePoints(RouteDescription routeDescription, MapData mapData) {
+    final List<Territory> territories = routeDescription.getRoute().getAllTerritories();
+    final int numTerritories = territories.size();
+    final Point[] points = new Point[numTerritories];
+    for (int i = 0; i < numTerritories; i++) {
+      points[i] = mapData.getCenter(territories.get(i));
     }
+    if (routeDescription.getStart() != null) {
+      points[0] = routeDescription.getStart();
```

Comment:
Failing the above,  a comment to say why we are overwriting the first value would be useful.